[Wrangler] Add docs for the delete command (#6792)
cloudflare/cloudflare-docs: content/workers/wrangler/commands.md

---

## delete

> Delete your Worker, and all associated Cloudflare developer platform resources.

Author: ```suggestion
Delete your Worker and all associated Cloudflare developer platform resources.
```

> Delete your Worker, its bindings, and any associated Workers Sites or asset KV namespaces from Cloudflare.

Reviewer: Are DOs included in the list of deleted things? If so maybe we should change wording to 'all associated Cloudflare developer platform resources'? I am not sure how much gets deleted with the Worker. But if this is all, then good as is.

Author: Good catch

> Delete your Worker, its bindings, and any associated Workers Sites or asset KV namespaces from Cloudflare.

Author: ```suggestion
Delete your Worker, and all associated Cloudflare developer platform resources.
```

```sh
$ wrangler delete [SCRIPT] [OPTIONS]
```

{{<definitions>}}

- `SCRIPT` {{<type>}}string{{</type>}}
  - The path to an entry point for your Worker.
- `--name` {{<type>}}string{{</type>}}
  - Name of the Worker.
- `--env` {{<type>}}string{{</type>}}
  - Perform on a specific environment.
- `--dry-run` {{<type>}}boolean{{</type>}} {{<prop-meta>}}(default: false){{</prop-meta>}}

>   - Don't actually delete the Worker. This is useful for testing the output of `wrangler delete`. 

Reviewer: ```suggestion
  - Do not actually delete the Worker. This is useful for testing the output of `wrangler delete`. 
```
